rest/src/main/scala/org/apache/gearpump/services/AppMasterService.scala

```diff
@@ -18,54 +18,58 @@
 
 package org.apache.gearpump.services
 
-import akka.actor.{Actor, ActorRef}
+import akka.actor.{ActorSystem, ActorRef}
 import akka.pattern.ask
-import com.wordnik.swagger.annotations._
-import org.apache.gearpump.cluster.MasterToAppMaster.{AppMasterData, AppMasterDataRequest}
-import org.apache.gearpump.util.Constants
+import org.apache.gearpump.cluster.AppMasterToMaster.AppMasterDataDetail
+import org.apache.gearpump.cluster.MasterToAppMaster.{AppMasterData, AppMasterDataDetailRequest, AppMasterDataRequest}
+import org.apache.gearpump.cluster.UserConfig
+import org.apache.gearpump.partitioner.Partitioner
+import org.apache.gearpump.streaming.{AppDescription, TaskDescription, DAG}
+import org.apache.gearpump.util.{Graph, Constants}
 import spray.http.StatusCodes
-import spray.routing
 import spray.routing.HttpService
 
 import scala.concurrent.{ExecutionContext, Future}
 import scala.util.{Failure, Success, Try}
 
-@Api(value = "/appmaster", description = "AppMaster Info.")
-class AppMasterServiceActor(val master:ActorRef) extends Actor with AppMasterService   {
-
-
-  def actorRefFactory = context
-  def receive = runRoute(readRoute)
-}
-
-@Api(value = "/appmaster", description = "AppMaster Info.")
-trait AppMasterService extends HttpService {
-  import org.apache.gearpump.services.AppMasterProtocol._
-  import spray.httpx.SprayJsonSupport._
-
-  implicit val ec: ExecutionContext = actorRefFactory.dispatcher
-  implicit val timeout = Constants.FUTURE_TIMEOUT
+trait AppMasterService extends HttpService  {
+  import upickle._
   val master:ActorRef
+  implicit val system: ActorSystem
 
-  val routes = readRoute 
+  def dag(config: Option[UserConfig]): DAG = {
+    config match {
+      case Some(_config) =>
+        DAG(_config.getValue[Graph[TaskDescription, Partitioner]](AppDescription.DAG).getOrElse(Graph.empty[TaskDescription, Partitioner]))
+      case None =>
```

Comment:
Can we move this to AppDescription, by adding an implicit conversion?

I asked the same question before.

You said it won't work because it requires an implicit actor system. Why it is a problem? Cannot we provide an implicit actor system?